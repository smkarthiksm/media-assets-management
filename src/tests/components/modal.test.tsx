import '@testing-library/jest-dom';
import { renderWithProviders } from '../test-utility';
import ModalComponent from '../../components/modal/modal';
import { fireEvent, screen } from '@testing-library/react';

describe('ModalComponent', () => {
  test('should render', () => {
    const { container } = renderWithProviders(
      <ModalComponent
        closeButtonTitle="Test"
        handleCancel={() => jest.fn()}
        handleClose={() => jest.fn()}
        modalTitle="Test Title"
        open={true}
        size="sm"
        disableCloseButton={false}
      >
        Test Content
      </ModalComponent>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should trigger close function', () => {
    const mockFunction = jest.fn();
    const closeButtonTitle = 'Close';
    renderWithProviders(
      <ModalComponent
        closeButtonTitle={closeButtonTitle}
        handleCancel={() => jest.fn()}
        handleClose={mockFunction}
        modalTitle="Test Title"
        open={true}
        size="sm"
        disableCloseButton={false}
      >
        Test Content
      </ModalComponent>,
    );

    const closeButton = screen.getByRole('button', {
      name: closeButtonTitle,
    });
    fireEvent.click(closeButton);
    expect(mockFunction).toBeCalled();
  });

  test('should trigger cancel function', () => {
    const mockFunction = jest.fn();
    renderWithProviders(
      <ModalComponent
        closeButtonTitle="Close"
        handleCancel={mockFunction}
        handleClose={() => jest.fn()}
        modalTitle="Test Title"
        open={true}
        size="sm"
        disableCloseButton={false}
      >
        Test Content
      </ModalComponent>,
    );

    const cancelButton = screen.getByRole('button', {
      name: 'Cancel',
    });
    fireEvent.click(cancelButton);
    expect(mockFunction).toBeCalled();
  });
});
