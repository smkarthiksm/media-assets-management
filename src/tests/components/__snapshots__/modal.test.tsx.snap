// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent should render 1`] = `
<div
  aria-hidden="true"
/>
`;
